docs/contracts.rst

```diff
@@ -968,9 +968,12 @@ the base constructors. This can be done in two ways::
         function Base(uint _x) public { x = _x; }
     }
 
-    contract Derived is Base(7) {
-        function Derived(uint _y) Base(_y * _y) public {
-        }
+    contract Derived1 is Base(7) {
+        function Derived1(uint _y) public {}
+    }
+
+    contract Derived2 is Base {
+        function Derived2(uint _y) Base(_y * _y) public {}
```

Comment:
Should we also warn if the constructor has a parameter but it is not explicitly initialised?

Reply:
This is a good example where this is useful, namely if the parameter depends on the constructor parameter of the derived contract. I hope we already have a warning if the parameters are specified nowhere...

Reply:
> I hope we already have a warning if the parameters are specified nowhere...

No, it just defaults to the type default.

Reply:
It’s impossible to know whether the contract is deployable as it is or meant as a base contract, so if we enforce this, the contract that specifies a base contract must also provide an explicit, though overridable, argument for all base contract parameters.

Reply:
One more reason to have an explicit `abstract` marker.

Reply:
Constructors could be chained through the abstract contracts explicitly, but that is a breaking change.